src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -928,6 +929,7 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     -->
   <PropertyGroup>
     <CoreBuildDependsOn>
+      $(CoreBuildDependsOn);
```

Comment:
Not as bad as `RebuildDependsOn`, but in the same concern area: whether a user target runs before or after `Compile` is likely super important, so to decide they go before it seems arbitrary.
If some or most of the sub-targets below have their own DependsOn properties, they would probably make better targets to tack on others to.